### PR TITLE
Try #4 by me to fix preferences

### PR DESCRIPTION
**Pull request by Intigracy**
Loops loading save until you load **if you first successfully load preferences.**
Moves the while(!speciesinit) to before your save loading/random character generating, since loading preferences doesn't need it and is being delayed for no reason.

Justification:
## WARNING: 231: datum/preferences/load_save_sqlite has returned in code/modules/client/preferences_savefile.dm at line 231 src: /datum/preferences usr: [whocaresabouttheckey]

https://github.com/Intigracy/vgstation13/blob/8266b23aa2ecaf0f1ccb46851b19be68ca2d65c9/code/modules/client/preferences_savefile.dm#L231

This check implies they are failing to load a save because they don't have a save to load, but that's obviously false.

This happens in the runtimes since the WARNING message was added in my PR earlier today.
The failure doesn't happen for the same people every round, which leads me to once again assume it's bottlenecking. If it is, this PR will definitively fix the issue.

TL;DR BOTTLENECKING

### CODE REVIEW

**Comment by gbasood** [comment]
Not sure if this'll break loading species in total or nto but I DONT EVEN KNOW ANYMORE

**Comment by clusterfack** [comment]
The idea isn't wrong but you should really proofread your code better so you don't cause infinite loops. Also as I have said using spawn to call the same function looks really really stupid its better to just use a while loop.

**Comment by Intigracy** [comment]
```
while(!saveloaded)
  if(load_save_sqlite(theckey, theclient, theslot))
    saveloaded = 1
    return
  else
    sleep(10)
```

Like this?

Updated.

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -186,26 +186,29 @@ var/const/MAX_SAVE_SLOTS = 8
...
-					if(!C)
-						WARNING("[__LINE__]: client C is null for [theckey].")
-					if(load_save_sqlite(theckey, C, default_slot) && C)
-						saveloaded = 1
-						return
-					else
-						world.log << "[theckey] failed loading save slot."
-				else
-					world.log << "[theckey] failed loading preferences."
+					while(!speciesinit)
```
Comment (gbasood) [inline code/modules/client/preferences.dm:192]: What's going on with the indentation here? The while block doesn't have anything inside it
Also if you put a while inside a while you will end up with double loops which is not great, maybe make try_load_save_sqlite not a while loop if you want a while loop in this block

Comment (Intigracy) [inline code/modules/client/preferences.dm:192]: oh I forgot to tab it.
the sleep is supposed to be in the while but the try isn't.

**Comment by Intigracy** [comment]
I've added a failsafe in case someone fails to load their save 100 times, which currently will take 100 seconds.

File: code/modules/client/preferences.dm [hunk 2]
```diff
@@ -186,26 +186,41 @@ var/const/MAX_SAVE_SLOTS = 8
...
+			while(!speciesinit)
+				sleep(1)
 			randomize_appearance_for()
 			real_name = random_name(gender)
 			save_character_sqlite(theckey, C, default_slot)
 			saveloaded = 1
 
+/datum/preferences/proc/try_load_save_sqlite(var/theckey, var/theclient, var/theslot)
+	var/attempts = 0
+	while(!saveloaded)
```
Comment (clusterfack) [inline code/modules/client/preferences.dm]:
```
while(!load_save_sqlite(theckey, theclient, theslot && attempts < 100)
    sleep(10)
    attempts++
saveloaded = 1

Comment (Intigracy) [inline code/modules/client/preferences.dm]: can't compile, you forgot a )
:^)

Comment (clusterfack) [inline code/modules/client/preferences.dm]: I'm the one who told you that
